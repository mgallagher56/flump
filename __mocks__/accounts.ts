import type { Account } from '~/containers/accounts/types';

const created_at = '2023-07-23T16:10:13.224361+00:00';
const user_id = 'a0eebc99-9c0b-4ef8-bb6d-6bb9bd380a11';

export default [
  {
    id: 'acdbb355-88b0-4973-85e4-debdc4f11231',
    created_at,
    name: 'Revolut',
    type: 'Current',
    user_id
  },
  {
    id: 'af2aa3d1-8524-42d8-b999-0d74e4b1a794',
    created_at,
    name: 'Revolut',
    type: 'Saving',
    user_id
  },
  {
    id: 'f9266923-85aa-4b6c-8af0-a79dcc838800',
    created_at,
    name: 'Starling',
    type: 'Saving',
    user_id
  },
  {
    id: '7c587659-3ca6-4b1f-9099-737901c14f92',
    created_at,
    name: 'Chip',
    type: 'Saving',
    user_id
  },
  {
    id: '17984b6e-5e94-4bfa-87d0-d7fc67342b82',
    created_at,
    name: 'Amex',
    type: 'Credit',
    user_id
  },
  {
    id: 'a785b22b-c168-4d24-9dbd-9b7ebb4d2b6b',
    created_at,
    name: 'Barclaycard',
    type: 'Credit',
    user_id
  },
  {
    id: '29078eeb-d28b-4e4b-a684-32c4e717675c',
    created_at,
    name: 'Barclays',
    type: 'Mortgage',
    user_id
  },
  {
    id: 'a6136944-6bd9-479c-b31d-e91e77b45baa',
    created_at,
    name: 'Halifax',
    type: 'Loan',
    user_id
  },
  {
    id: 'f51772ed-0417-4620-aa4b-4ff4fe88c125',
    created_at,
    name: 'John Smith',
    type: 'Owed',
    user_id
  },
  {
    id: 'ec289633-3a61-4c41-ad06-4a136d37e29f',
    created_at: '2023-07-23T22:00:27.082678+00:00',
    name: '',
    type: '',
    user_id
  },
  {
    id: '4deafe69-2c47-4935-9dd9-30d9ad620898',
    created_at: '2023-07-23T22:11:02.314859+00:00',
    name: '',
    type: '',
    user_id
  },
  {
    id: 'f0b60bd5-a098-489b-b251-3ec6b24cae7c',
    created_at: '2023-07-24T08:16:51.83913+00:00',
    name: '',
    type: '',
    user_id
  },
  {
    id: '1d0fa360-0bae-4bb6-bd8d-8fee9c11d116',
    created_at: '2023-07-24T08:56:03.591809+00:00',
    name: 'sfsd',
    type: 'sdfsdfsdf',
    user_id
  },
  {
    id: '1b0914c2-805f-46a9-8b36-0185119d5120',
    created_at,
    name: 'Starling',
    type: 'Current',
    user_id
  }
] as Account[];
