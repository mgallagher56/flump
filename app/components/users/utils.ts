import type { SupabaseClient } from '@supabase/supabase-js';
import type { TFunction } from 'i18next';

export enum SignUpActionEnum {
  LOGIN = 'logIn',
  SIGNUP = 'signUp'
}

export const getSignUpButtonText = (t: TFunction, isShowLogin: boolean, isMagicLink: boolean) => {
  if (isMagicLink) return t('sendMagicLink');
  if (isShowLogin) return t(SignUpActionEnum.LOGIN);
  return t(SignUpActionEnum.SIGNUP);
};

export const getInfoMessage = (t: TFunction, isUserRegistered: boolean, isMagicLink: boolean) => {
  if (isMagicLink) return t('magicLinkSent');
  if (isUserRegistered) return t('clickAgainToLogIn');
  return null;
}

export const getSubmitUserAuthAction = async ({
  supabaseClient,
  isShowLogin,
  isMagicLink,
  formInput
}: {
  supabaseClient: SupabaseClient;
  isShowLogin: boolean;
  isMagicLink: boolean;
  formInput: { email: string; password: string };
}) => {
  if (isMagicLink)
    return await supabaseClient.auth.signInWithOtp({
      email: formInput.email,
      options: {
        emailRedirectTo: window.location.origin + '/auth/callback'
      }
    });
  if (isShowLogin) return await supabaseClient.auth.signInWithPassword(formInput);
  return await supabaseClient.auth.signUp(formInput);
};
