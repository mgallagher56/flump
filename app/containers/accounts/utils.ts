import type { Account } from './types';

export enum AccountTypeEnum {
  CURRENT = 'Current',
  SAVING = 'Saving',
  MORTGAGE = 'Mortgage',
  LOAN = 'Loan',
  OWED = 'Owed'
}

export const isAccountTypeValid = (accountType: AccountTypeEnum, accounts: Account[]) =>
  !!accounts?.map(({ type }) => type).some((type) => type === accountType);
