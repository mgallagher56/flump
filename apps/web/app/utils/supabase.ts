import { createBrowserClient, createServerClient, parseCookieHeader, serializeCookieHeader } from '@supabase/ssr';

import type { Database } from 'db_types';

declare global {
  interface Window {
    env: {
      SUPABASE_URL: string;
      SUPABASE_ANON_KEY: string;
    };
  }
}

const isServer = typeof window === 'undefined';

export const getSupaBaseUrl = (isServer: boolean): string =>
  isServer ? process.env.SUPABASE_URL : window?.env?.SUPABASE_URL;
export const getSupabaseAnonKey = (isServer: boolean): string =>
  isServer ? process.env.SUPABASE_ANON_KEY : window?.env?.SUPABASE_ANON_KEY;

const supabase = createBrowserClient<Database>(getSupaBaseUrl(isServer), getSupabaseAnonKey(isServer));

export const createSupaBaseServerClient = (request: Request) => {
  const headers = new Headers();

  return createServerClient<Database>(getSupaBaseUrl(isServer), getSupabaseAnonKey(isServer), {
    cookies: {
      getAll() {
        return parseCookieHeader(request.headers.get('Cookie') ?? '');
      },
      setAll(cookiesToSet) {
        cookiesToSet.forEach(({ name, value, options }) => {
          headers.append('Set-Cookie', serializeCookieHeader(name, value, options));
        });
      }
    }
  });
};

export default supabase;
