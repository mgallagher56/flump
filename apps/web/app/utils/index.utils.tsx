import type { ReactElement } from 'react';

import type { TFunction } from 'i18next';
import FLPBox from '~/components/core/structure/FLPBox';
import type { TabData } from '~/components/core/tabs/types';
import FLPHeading from '~/components/core/typography/FLPHeading';
import type { Employee } from '~/routes/_index';

import { FallbackEnums } from './utils';

const { NA } = FallbackEnums;

export const getTabsData = (employees: Partial<Employee>[], t: TFunction): TabData[] => [
  {
    label: t('featureNum', { num: 1 }),
    value: 'feature-1',
    children: (
      <main>
        <FLPBox>
          <FLPHeading as="h1">{t('featureNum', { num: 1 })}</FLPHeading>
          {employees.map(
            (employee, index): ReactElement => (
              <FLPBox key={employee.id ?? index} style={{ display: 'flex', gap: '.5rem' }}>
                <h3>{`${employee.name ?? NA} - ${employee.department ?? NA}`}</h3>
              </FLPBox>
            )
          )}
        </FLPBox>
      </main>
    )
  },
  {
    label: t('featureNum', { num: 2 }),
    value: 'feature 2',
    children: (
      <main>
        <FLPBox>
          <FLPHeading as="h1">{t('featureNum', { num: 2 })}</FLPHeading>
        </FLPBox>
      </main>
    )
  },
  {
    label: t('featureNum', { num: 3 }),
    value: 'feature-3',
    children: (
      <main>
        <FLPBox>
          <FLPHeading as="h1">{t('featureNum', { num: 3 })}</FLPHeading>
        </FLPBox>
      </main>
    )
  }
];
