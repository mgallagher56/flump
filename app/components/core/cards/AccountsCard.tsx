import { type FC, useCallback } from 'react';

import type { CardProps } from '@chakra-ui/react';
import { useLoaderData, useNavigate, useRevalidator } from '@remix-run/react';
import { useTranslation } from 'react-i18next';
import type { AccountTypeEnum } from '~/containers/accounts/utils';
import AddEditAccountsDialogBtn from '~/containers/dialogs/addEditAccountsDialog.tsx/AddEditAccountsDialog';
import type { loader } from '~/root';
import supabase from '~/utils/supabase';

import FLPButton from '../buttons/FLPButton';
import FLPButtonGroup from '../buttons/FLPButtonGroup';
import FLPBox from '../structure/FLPBox';
import FLPText from '../typography/FLPText';
import FLPCard from './FLPCard';

interface AccountsCardProp extends Omit<CardProps, 'title'> {
  accountId: string;
  name: string;
  type: AccountTypeEnum;
}

const AccountsCard: FC<AccountsCardProp> = ({ accountId, name, type }) => {
  const { t } = useTranslation();
  const navigate = useNavigate();
  const { user } = useLoaderData<typeof loader>();
  const { revalidate } = useRevalidator();

  const handleRemoveAccount = useCallback(
    async (e) => {
      e.stopPropagation();
      await supabase.from('accounts').delete().eq('user_id', user.id).eq('id', accountId);
      revalidate();
    },
    [accountId, revalidate, user.id]
  );

  const onCardClick = useCallback(() => navigate(`/app/accounts/${accountId}`), [accountId, navigate]);

  return (
    <FLPCard
      id={accountId}
      title={name}
      padding={5}
      marginY={5}
      width="250px"
      height="300px"
      display="flex"
      direction="column"
      justifyContent="space-between"
      gap={2}
      onClick={onCardClick}
    >
      <FLPBox display="flex" gap="3" flexDirection="column">
        <FLPText fontWeight="bold">{name}</FLPText>
        <FLPText>{`${type} ${t('account')}`}</FLPText>
      </FLPBox>
      <FLPButtonGroup justifyContent="flex-end" zIndex="10">
        <AddEditAccountsDialogBtn accountId={accountId} isEditAccount={true} />
        <FLPButton colorScheme="red" onClick={handleRemoveAccount}>
          {t('remove')}
        </FLPButton>
      </FLPButtonGroup>
    </FLPCard>
  );
};

export default AccountsCard;
