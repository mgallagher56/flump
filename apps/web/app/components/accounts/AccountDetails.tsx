import { Fragment, useCallback } from 'react';
import type { ChangeEvent, FC } from 'react';

import { Stack } from '@chakra-ui/react';
import { useTranslation } from 'react-i18next';
import { useLoaderData, useRevalidator } from 'react-router';
import FLPButton from '~/components/core/buttons/FLPButton';
import FLPInput from '~/components/core/inputs/input/FLPInput';
import FLPHeading from '~/components/core/typography/FLPHeading';
import FLPText from '~/components/core/typography/FLPText';
import type { loader } from '~/routes/app.accounts.$account';
import supabase from '~/utils/supabase';

import { accountDetailDisplayStyles } from './AccountDetailsStyles';

interface AccountDetailsProps {
  onInputChange: (event: ChangeEvent<HTMLInputElement>) => void;
  editedValues: { [key: string]: { [key: string]: string } };
  isEditMode: boolean;
}

const AccountDetails: FC<AccountDetailsProps> = ({ onInputChange, editedValues, isEditMode }) => {
  const { t } = useTranslation();
  const { account, accountDetails } = useLoaderData<typeof loader>();
  const { revalidate } = useRevalidator();

  const onRemoveYear = useCallback(
    async (year: number) => {
      await supabase.from('account_details').delete().eq('account_id', account.id).eq('year', year).select();
      revalidate();
    },
    [account.id, revalidate]
  );

  return (
    <Stack flexDirection="column" gap={5} overflow="auto">
      {accountDetails
        .map(({ year }: { year?: number }) => year)
        .filter(
          (
            value: number,
            index: number,
            self: {
              indexOf: (arg0: number) => number;
            }
          ) => self.indexOf(value) === index
        )
        .map((year: number) => (
          <Fragment key={year}>
            <Stack alignItems="center" flexDirection="row">
              <FLPHeading as="h3" size="lg">
                {year}
              </FLPHeading>
              <FLPButton colorScheme="red" size="xs" variant="outline" onClick={() => onRemoveYear(year)}>
                {t('deleteYear')}
              </FLPButton>
            </Stack>
            <div className={accountDetailDisplayStyles}>
              {accountDetails.map(
                ({ month, year: currentYear, value }: { month?: number; year?: number; value?: number }) => {
                  if (year !== currentYear) return null;
                  const monthName = new Date(0, month - 1).toLocaleString('default', { month: 'long' });
                  return (
                    <div key={`${year}-${month}`}>
                      <FLPText>{monthName}</FLPText>
                      {isEditMode ? (
                        <FLPInput
                          isLabelHidden
                          data-month={month}
                          data-year={year}
                          label={`${year}-${monthName}`}
                          name="amount"
                          placeholder="0.00"
                          type="number"
                          value={editedValues?.[year]?.[month] ?? value}
                          width={100}
                          onChange={onInputChange}
                        />
                      ) : (
                        <FLPText>
                          {Intl.NumberFormat('en-GB', { style: 'currency', currency: 'GBP' })
                            .format(value)
                            .slice(0, -3)}
                        </FLPText>
                      )}
                    </div>
                  );
                }
              )}
            </div>
          </Fragment>
        ))}
    </Stack>
  );
};

export default AccountDetails;
