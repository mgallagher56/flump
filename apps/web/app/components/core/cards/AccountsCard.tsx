import { type FC, useCallback, useMemo } from 'react';

import {
  CardBody,
  CardFooter,
  CardHeader,
  type CardProps,
  Stack,
  Stat,
  StatArrow,
  StatHelpText,
  StatLabel,
  StatNumber
} from '@chakra-ui/react';
import { useLoaderData, useNavigate, useRevalidator } from '@remix-run/react';
import { useTranslation } from 'react-i18next';
import AccountDetailChart from '~/components/charts/AccountDetailChart';
import FLPButton from '~/components/core/buttons/FLPButton';
import FLPButtonGroup from '~/components/core/buttons/FLPButtonGroup';
import FLPHeading from '~/components/core/typography/FLPHeading';
import AddEditAccountsDialogBtn from '~/components/dialogs/addEditAccountsDialog.tsx/AddEditAccountsDialog';
import type { AccountDetail } from '~/containers/accounts/types';
import type { AccountTypeEnum } from '~/containers/accounts/utils';
import type { loader } from '~/routes/app.accounts._index';
import { monthYearSort } from '~/utils/accounts';
import supabase from '~/utils/supabase';
import { currentMonth, currentYear } from '~/utils/utils';

import FLPCard from './FLPCard';

interface AccountsCardProp extends Omit<CardProps, 'title'> {
  accountId: string;
  name: string;
  type: AccountTypeEnum;
}

const AccountsCard: FC<AccountsCardProp> = ({ accountId, name, type }) => {
  const { t } = useTranslation();
  const navigate = useNavigate();
  const {
    user,
    accountDetails = []
  }: {
    user?: { id?: string };
    accountDetails?: AccountDetail[];
  } = useLoaderData<typeof loader>();
  const { revalidate } = useRevalidator();

  const sortedAccountDetails = useMemo(() => {
    return accountDetails.filter((account) => account.account_id === accountId).sort(monthYearSort);
  }, [accountDetails, accountId]);

  const currentMonthIndex = sortedAccountDetails.findIndex(
    (account) => account.month === currentMonth && account.year === currentYear
  );

  const accountDetailYear = useMemo(() => {
    if (sortedAccountDetails.length === 12) return sortedAccountDetails;

    return sortedAccountDetails.slice(currentMonthIndex - 11, currentMonthIndex + 1);
  }, [currentMonthIndex, sortedAccountDetails]);

  const accountBalance = accountDetailYear[11]?.value;
  const prevAccountBalance = accountDetailYear[10]?.value;
  const secondPreviousAccountBalance = accountDetailYear[9]?.value;

  const handleRemoveAccount = useCallback(
    async (e: { stopPropagation: () => void }) => {
      e.stopPropagation();
      await supabase.from('accounts').delete().eq('user_id', user.id).eq('id', accountId);
      revalidate();
    },
    [accountId, revalidate, user.id]
  );

  const onCardClick = useCallback(() => navigate(`/app/accounts/${accountId}`), [accountId, navigate]);

  return (
    <FLPCard
      direction="column"
      display="flex"
      gap={3}
      height="325px"
      id={accountId}
      justifyContent="space-between"
      marginY={5}
      padding={5}
      title={name}
      width="250px"
      onClick={onCardClick}
    >
      <CardHeader display="flex" flexDirection="column" gap={1} padding={0}>
        <FLPHeading as="h5" color="grey.500" size="xs">{`${type} ${t('account')}`}</FLPHeading>
        <FLPHeading as="h4" color="blue.500" fontWeight="bold" size="lg">
          {name}
        </FLPHeading>
      </CardHeader>
      <Stack alignItems="center" direction="row" display="flex" justifyContent="space-between">
        {!!prevAccountBalance && (
          <Stat size="xs">
            <StatLabel>{`${t('previous')}:`}</StatLabel>
            <StatNumber>
              {Intl.NumberFormat('en-GB', { style: 'currency', currency: 'GBP', maximumFractionDigits: 0 }).format(
                prevAccountBalance
              )}
            </StatNumber>
            {!!secondPreviousAccountBalance && (
              <StatHelpText>
                {secondPreviousAccountBalance > prevAccountBalance ? (
                  <StatArrow type="decrease" />
                ) : (
                  <StatArrow type="increase" />
                )}
                {`${Math.round(
                  ((prevAccountBalance - secondPreviousAccountBalance) / secondPreviousAccountBalance) * 100
                )}%`}
              </StatHelpText>
            )}
          </Stat>
        )}
        {!!accountBalance && (
          <Stat size="xs">
            <StatLabel>{`${t('current')}:`}</StatLabel>
            <StatNumber>
              {Intl.NumberFormat('en-GB', {
                style: 'currency',
                currency: 'GBP',
                maximumFractionDigits: 0
              }).format(accountBalance)}
            </StatNumber>
            {!!prevAccountBalance && (
              <StatHelpText>
                {prevAccountBalance > accountBalance ? <StatArrow type="decrease" /> : <StatArrow type="increase" />}
                {`${Math.round(((accountBalance - prevAccountBalance) / prevAccountBalance) * 100)}%`}
              </StatHelpText>
            )}
          </Stat>
        )}
      </Stack>
      <CardBody justifyContent="flex-end" padding={0}>
        {!!accountDetailYear.length && <AccountDetailChart accountDetails={accountDetailYear} />}
      </CardBody>
      <CardFooter justifyContent="flex-end" padding={0}>
        <FLPButtonGroup justifyContent="flex-end" zIndex="10">
          <AddEditAccountsDialogBtn accountId={accountId} btnSize="sm" isEditAccount={true} />
          <FLPButton colorScheme="red" size="sm" onClick={handleRemoveAccount}>
            {t('delete')}
          </FLPButton>
        </FLPButtonGroup>
      </CardFooter>
    </FLPCard>
  );
};

export default AccountsCard;
