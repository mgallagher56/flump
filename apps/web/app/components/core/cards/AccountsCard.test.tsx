import { fireEvent, render } from '@testing-library/react';
import { vi } from 'vitest';
import { AccountTypeEnum } from '~/containers/accounts/utils';
import { currentMonth, currentYear } from '~/utils/utils';

import mockUser from '__mocks__/user';

import AccountsCard from './AccountsCard';

const mocks = vi.hoisted(() => ({
  mockUseLoaderData: vi.fn(),
  mockUseRevalidator: vi.fn(() => ({ revalidate: vi.fn() })),
  mockUseNavigate: () => vi.fn(),
  mockFrom: vi.fn(() => ({
    delete: () => ({
      eq: () => ({
        eq: () => ({})
      })
    })
  }))
}));

vi.mock('react-router', async () => {
  const actual: Record<string, unknown> = await vi.importActual('react-router');
  return {
    ...actual,
    useLoaderData: mocks.mockUseLoaderData,
    useRevalidator: mocks.mockUseRevalidator,
    useNavigate: mocks.mockUseNavigate
  };
});

vi.mock('app/utils/supabase', () => ({
  default: {
    from: mocks.mockFrom
  }
}));

vi.mock('app/utils/utils', () => ({
  currentMonth: 12,
  currentYear: 2023
}));

const getAdjustedValue = (index: number) => {
  if (currentMonth - 1 === index) return `${index - 2}000`;
  if (currentMonth - 3 === index) return `${index + 5}000`;

  return `${index}000`;
};

describe('<AccountsCard />', () => {
  test('it renders an AccountsCard component with title as expected', () => {
    mocks.mockUseLoaderData.mockReturnValue({
      user: mockUser,
      accountDetails: Array.from({ length: 12 }, (_, i) => {
        return {
          id: i,
          account_id: '123456',
          month: i + 1,
          year: currentYear,
          value: parseInt(`${i + 1}000}`)
        };
      })
    });

    const { baseElement } = render(
      <AccountsCard accountId={'123456'} name="My curent account" type={AccountTypeEnum.CURRENT} />
    );
    expect(baseElement).toMatchSnapshot();
  });
});

describe('<AccountsCard with increasing values', () => {
  test('it calls supabase.delete when delete button is clicked', () => {
    mocks.mockUseLoaderData.mockReturnValueOnce({
      user: mockUser,
      accountDetails: Array.from({ length: 13 }, (_, i) => {
        return {
          id: i,
          account_id: '123456',
          month: i + 1,
          year: currentYear,
          value: parseInt(getAdjustedValue(i))
        };
      })
    });

    const { baseElement, getByText } = render(
      <AccountsCard accountId={'123456'} name="My curent account" type={AccountTypeEnum.CURRENT} />
    );
    const deleteButton = getByText('delete');
    fireEvent.click(deleteButton);
    expect(mocks.mockFrom).toBeCalledWith('accounts');
    expect(baseElement).toMatchSnapshot();
  });
});
