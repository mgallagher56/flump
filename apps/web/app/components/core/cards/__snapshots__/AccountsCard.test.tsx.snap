// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<AccountsCard /> > it renders an AccountsCard component with title as expected 1`] = `
<body>
  <div>
    <div
      class="chakra-card pos_relative trs_all_0.3s_ease-in-out bd_1px_solid_token(colors.gray.500)! [&:hover]:bx-sh_2px_5px_8px_0px_token(colors.gray.500)! [&:hover]:trf_translate3d(-2px,_-5px,_-5px) [&:hover]:cursor_pointer css-11joc2e"
      id="123456"
      title="My curent account"
    >
      <div
        class="chakra-card__header css-m9wzl5"
      >
        <h5
          class="chakra-heading css-khlt38"
        >
          Current account
        </h5>
        <h4
          class="chakra-heading css-svk9li"
        >
          My curent account
        </h4>
      </div>
      <div
        class="chakra-stack css-1y3f6ad"
      >
        <div
          class="chakra-stat css-1mbo1ls"
        >
          <dl>
            <dt
              class="chakra-stat__label css-0"
            >
              previous:
            </dt>
            <dd
              class="chakra-stat__number css-fvxuh5"
            >
              £11,000
            </dd>
            <dd
              class="chakra-stat__help-text css-0"
            >
              <span
                class="css-8b45rq"
              >
                increased by
              </span>
              <svg
                aria-hidden="true"
                class="chakra-icon css-m17ft2"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12.8,5.4c-0.377-0.504-1.223-0.504-1.6,0l-9,12c-0.228,0.303-0.264,0.708-0.095,1.047 C2.275,18.786,2.621,19,3,19h18c0.379,0,0.725-0.214,0.895-0.553c0.169-0.339,0.133-0.744-0.095-1.047L12.8,5.4z"
                  fill="currentColor"
                />
              </svg>
              10%
            </dd>
          </dl>
        </div>
        <div
          class="chakra-stat css-1mbo1ls"
        >
          <dl>
            <dt
              class="chakra-stat__label css-0"
            >
              current:
            </dt>
            <dd
              class="chakra-stat__number css-fvxuh5"
            >
              £12,000
            </dd>
            <dd
              class="chakra-stat__help-text css-0"
            >
              <span
                class="css-8b45rq"
              >
                increased by
              </span>
              <svg
                aria-hidden="true"
                class="chakra-icon css-m17ft2"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12.8,5.4c-0.377-0.504-1.223-0.504-1.6,0l-9,12c-0.228,0.303-0.264,0.708-0.095,1.047 C2.275,18.786,2.621,19,3,19h18c0.379,0,0.725-0.214,0.895-0.553c0.169-0.339,0.133-0.744-0.095-1.047L12.8,5.4z"
                  fill="currentColor"
                />
              </svg>
              9%
            </dd>
          </dl>
        </div>
      </div>
      <div
        class="chakra-card__body css-ttziis"
      >
        <div
          class="recharts-responsive-container"
          style="width: 100%; height: 95%; min-width: 0;"
        />
      </div>
      <div
        class="chakra-card__footer css-eldtq0"
      >
        <div
          class="chakra-button__group css-vhnxbm"
          data-orientation="horizontal"
          role="group"
        >
          <button
            class="chakra-button css-kn8t1u"
            type="button"
          >
            edit
          </button>
          <button
            class="chakra-button css-kn8t1u"
            type="button"
          >
            delete
          </button>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`<AccountsCard with increasing values > it calls supabase.delete when delete button is clicked 1`] = `
<body>
  <div>
    <div
      class="chakra-card pos_relative trs_all_0.3s_ease-in-out bd_1px_solid_token(colors.gray.500)! [&:hover]:bx-sh_2px_5px_8px_0px_token(colors.gray.500)! [&:hover]:trf_translate3d(-2px,_-5px,_-5px) [&:hover]:cursor_pointer css-11joc2e"
      id="123456"
      title="My curent account"
    >
      <div
        class="chakra-card__header css-m9wzl5"
      >
        <h5
          class="chakra-heading css-khlt38"
        >
          Current account
        </h5>
        <h4
          class="chakra-heading css-svk9li"
        >
          My curent account
        </h4>
      </div>
      <div
        class="chakra-stack css-1y3f6ad"
      >
        <div
          class="chakra-stat css-1mbo1ls"
        >
          <dl>
            <dt
              class="chakra-stat__label css-0"
            >
              previous:
            </dt>
            <dd
              class="chakra-stat__number css-fvxuh5"
            >
              £10,000
            </dd>
            <dd
              class="chakra-stat__help-text css-0"
            >
              <span
                class="css-8b45rq"
              >
                decreased by
              </span>
              <svg
                aria-hidden="true"
                class="chakra-icon css-19fbr80"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M21,5H3C2.621,5,2.275,5.214,2.105,5.553C1.937,5.892,1.973,6.297,2.2,6.6l9,12 c0.188,0.252,0.485,0.4,0.8,0.4s0.611-0.148,0.8-0.4l9-12c0.228-0.303,0.264-0.708,0.095-1.047C21.725,5.214,21.379,5,21,5z"
                  fill="currentColor"
                />
              </svg>
              -29%
            </dd>
          </dl>
        </div>
        <div
          class="chakra-stat css-1mbo1ls"
        >
          <dl>
            <dt
              class="chakra-stat__label css-0"
            >
              current:
            </dt>
            <dd
              class="chakra-stat__number css-fvxuh5"
            >
              £9,000
            </dd>
            <dd
              class="chakra-stat__help-text css-0"
            >
              <span
                class="css-8b45rq"
              >
                decreased by
              </span>
              <svg
                aria-hidden="true"
                class="chakra-icon css-19fbr80"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M21,5H3C2.621,5,2.275,5.214,2.105,5.553C1.937,5.892,1.973,6.297,2.2,6.6l9,12 c0.188,0.252,0.485,0.4,0.8,0.4s0.611-0.148,0.8-0.4l9-12c0.228-0.303,0.264-0.708,0.095-1.047C21.725,5.214,21.379,5,21,5z"
                  fill="currentColor"
                />
              </svg>
              -10%
            </dd>
          </dl>
        </div>
      </div>
      <div
        class="chakra-card__body css-ttziis"
      >
        <div
          class="recharts-responsive-container"
          style="width: 100%; height: 95%; min-width: 0;"
        />
      </div>
      <div
        class="chakra-card__footer css-eldtq0"
      >
        <div
          class="chakra-button__group css-vhnxbm"
          data-orientation="horizontal"
          role="group"
        >
          <button
            class="chakra-button css-kn8t1u"
            type="button"
          >
            edit
          </button>
          <button
            class="chakra-button css-kn8t1u"
            type="button"
          >
            delete
          </button>
        </div>
      </div>
    </div>
  </div>
</body>
`;
